### ISSUE TITLE
Unable to launch/install on Docker without root privileges

### ISSUE THREAD

**Issue by blakebyrnes** [unblocked-web/agent#16]
If you try to install in Docker, there are issues with permissions to the apt-installer file. User tried to use some combo of:
```RUN cd /app/agent && yarn \
    && $(npx install-browser-deps) \
    && groupadd -r agent && useradd -r -g agent -G audio,video agent \
    && mkdir -p /home/agent/Downloads \
    && mkdir -p /home/agent/.cache \
    && chown -R agent:agent /home/agent \
    && chown -R agent:agent /app/agent \
    && mv ~/.cache/agent /home/agent/.cache/ \
    && chmod 777 /tmp \
    && chmod -R 777 /home/agent/.cache/agent```

but still showing issues like the following if reassigned to a user that Agent runs as.

```022-09-23T19:53:40.512Z ERROR [hero-core/connections/ConnectionToHeroClient] ConnectionToClient.HandleRequestError { context: {}, sessionId: undefined, sessionName: undefined } BrowserLaunchError: EACCES: permission denied, open '/tmp/apt-install-chrome-dependencies.sh'
------BROWSER_CREATION_Q--------------------------
```

````BrowserLaunchError: DependenciesMissingError: Ulixee Server failed to launch Chrome - Some of the dependencies needed to run Chrome 98.0.4758.102 are not on your system!

You can resolve this by running the apt dependency installer at:
-------------------------------------------------

/tmp/apt-install-chrome-dependencies.sh

-------------------------------------------------

missing: libgobject-2.0.so.0, libglib-2.0.so.0, libnss3.so, libnssutil3.so, libsmime3.so, libnspr4.so, libatk-1.0.so.0, libatk-bridge-2.0.so.0, libcups.so.2, libgio-2.0.so.0, libdrm.so.2, libdbus-1.so.3, libx11.so.6, libxcomposite.so.1, libxdamage.so.1, libxext.so.6, libxfixes.so.3, libxrandr.so.2, libgbm.so.1, libxcb.so.1, libxkbcommon.so.0, libpango-1.0.so.0, libcairo.so.2, libasound.so.2, libatspi.so.0
. See server logs for details.```

**Comment by blakebyrnes** [comment]
There was no code change needed here, just some CHMODs in the Dockerfile. Example can be seen here: https://github.com/ulixee/ulixee/blob/4e2892ecc0ce378f792f7b56fe873f63ef85a82b/server/tools/docker/Dockerfile#L52

**Comment by GlenDC** [comment]
For you that might work, but I never the less still get:

```
BrowserLaunchError: Ulixee Server failed to launch Chrome - EACCES: permission denied, open '/tmp/apt-install-chrome-dependencies.sh'. See server logs for details.
```

The CMODS you were talking about I already had, so don't think that is the magic solution, given I was already doing those.

**Comment by GlenDC** [comment]
Only difference is that I do `yarn --forzen-lockfile` instead of:

```
yarn init -yp \
    && yarn add @ulixee/server@$VERSION \
    && $ADD_TO_INSTALL \
```

Instead I copy my own src code, and install using yarn with frozen lockfile.

So could there be an issue because of that? I would hope not..


Also note that the build process works, it's only when I run the container and need to open a hero conn. that I get that isuse @blakebyrnes .

**Comment by blakebyrnes** [comment]
@GlenDC, I have it running on a Docker that's published... I realized after that it must not be the chmods, but I do think it's an order of creating directories and owning them. Do you have your full dockerfile somewhere?

**Comment by blakebyrnes** [comment]
It also does require you to run the docker with `--user ulixee` and an seccomp profile.

**Comment by GlenDC** [comment]
what does `seccomp profile` mean?

**Comment by blakebyrnes** [comment]
It's a linux security profile. That's the name of it if you google it. Here are some of my notes for running it:
https://github.com/ulixee/ulixee/tree/main/server/tools/docker

**Comment by GlenDC** [comment]
Hmmm ok the `--user` does make sense. However I do not think this really matters.
Also why is this needed? Shouldn't it by default respect the `USER` command in Dockerfile?

Also even more, didn't we last time realise the real issue is because of that `/tmp` dep script which contains code like:

```
if [ ! -f "/home/bro/.cache/ulixee/chrome/105.0.5195.125/.validated" ]; then
  chown _apt /home/bro/.cache/ulixee/chrome/105.0.5195.125/install-dependencies.deb;
  apt-get update;
  apt install -y /home/bro/.cache/ulixee/chrome/105.0.5195.125/install-dependencies.deb;
  apt-get -y autoremove;
  touch /home/bro/.cache/ulixee/chrome/105.0.5195.125/.validated;
  chmod 777 /home/bro/.cache/ulixee/chrome/105.0.5195.125/.validated;
fi
```

which used `apt-get` install etc, and thus that is what requires sudo, no?

Also is seccomp _really_ needed?

**Comment by blakebyrnes** [comment]
> Hmmm ok the `--user` does make sense. However I do not think this really matters. Also why is this needed? Shouldn't it by default respect the `USER` command in Dockerfile?

That's a good question. I don't know... I assume you're not asking why we need to run as non-root?

> which used `apt-get` install etc, and thus that is what requires sudo, no?
I'm not sure what happened in your docker, but I DO know that I got it working, ran it on Digital Ocean and connected to it :). 

It seems like your installation is running and determining Chrome dependencies are not present. That seems to indicate that somehow you're NOT running the installation script during the build phase, but it's finding missing dependencies at runtime. I think I need to see your whole Dockerfile to know more.


> Also is seccomp _really_ needed?
Yes, in order to run as non-root, Chrome needs to be able to create Sandboxed users (or you can run with --no-sandbox). If you want to run with Chrome Sandboxed AND non-root, you need it.

**Comment by GlenDC** [comment]
>  I assume you're not asking why we need to run as non-root?

Correct

**Comment by GlenDC** [comment]
Would there actually be an issue if I in fact keep running as root forever,
like... can it be detected? As I honestly can live with the warning.

**Comment by GlenDC** [comment]
>  I think I need to see your whole Dockerfile to know more.

Not that I mind giving it to you, but you would honestly not be anything with it. Only difference is that I run my won project instead of the server dependency.

Besides that I am just installing my deps etc...

So all I can think is that somehow adding the hero server like you do, does install it, while installing it as a dependency for another project, does not install it? Is the only difference I can see that matters.

Most of my dockerfile is identical to yours, as that is the one I based myself from.

**Comment by blakebyrnes** [comment]
> Would there actually be an issue if I in fact keep running as root forever,
> like... can it be detected? As I honestly can live with the warning.

Not that I know of. It's more that non sandboxed Chrome is considered a "scripting security" risk, and running non-root docker apps is also considered a "more secure" option. I think they're both just about escaping out of your shell.

> Not that I mind giving it to you, but you would honestly not be anything with it. Only difference is that I run my won project instead of the server dependency.

Even if you choose to run as root, I just want to make sure I'm not overlooking something. I don't see how a lock file would matter here, but a different base image would, as would order of setting permissions and running the installer. 

> Most of my dockerfile is identical to yours, as that is the one I based myself from.

There were several tweaks that I made to the docker file in ulixee/ulixee to get it working properly. They were very minor, but it did have some subtle changes.

**Comment by blakebyrnes** [comment]
Actually, I just compared to the /agent one and not that much did change. The main things I could think of that might matter:
1. How you're running the docker. This might matter, not sure. I can experiment here
2. The base image changing and not including something that Chrome wants (which would be prompting that installer to try to manipulate the tmp file). I changed to node-16 slim.
3. Some minor order change in commands

**Comment by blakebyrnes** [comment]
I think I found something. Stay tuned. Found symlinks pointing back to root.

**Comment by blakebyrnes** [comment]
Ok, I just pushed a commit that should fix your issue. Had to direct browsers to output into a directory so that symlinks won't break from moving the cache directory.

```
RUN mkdir -p /ulixee/browsers
ENV BROWSERS_DIR /ulixee/browsers

RUN ...     && chmod 777 /ulixee/browsers \

```

**Comment by GlenDC** [comment]
> It also does require you to run the docker with `--user ulixee` and an seccomp profile.

Doesn't seem like you need to use `--user` when running. Tried running without and I can see with `whomai` that I am the user which I set via `USER`. So all is good without the explicit need to define it like this.

**Comment by GlenDC** [comment]
Despite the addition of the browser I still get the same issue:

```
2022-10-27T16:50:35.264Z ERROR [hero-core/connections/ConnectionToHeroClient] ConnectionToClient.HandleRequestError { context: {}, sessionId: undefined, sessionName: undefined } BrowserLaunchError: DependenciesMissingError: Some of the dependencies needed to run Chrome 105.0.5195.125 are not on your system!

You can resolve this by running the apt dependency installer at:
-------------------------------------------------

/tmp/apt-install-chrome-dependencies.sh

-------------------------------------------------
```

Here is my dockerfile:

```Dockerfile
FROM node:16-slim as base

# Fonts
RUN echo "deb http://httpredir.debian.org/debian buster main contrib non-free" > /etc/apt/sources.list \
    && echo "deb http://httpredir.debian.org/debian buster-updates main contrib non-free" >> /etc/apt/sources.list \
    && echo "deb http://security.debian.org/ buster/updates main contrib non-free" >> /etc/apt/sources.list \
    && echo "ttf-mscorefonts-installer msttcorefonts/accepted-mscorefonts-eula select true" | debconf-set-selections \
    && apt-get update \
    && apt-get install -y \
        fonts-arphic-ukai \
        fonts-arphic-uming \
        fonts-ipafont-mincho \
        fonts-thai-tlwg \
        fonts-kacst \
        fonts-ipafont-gothic \
        fonts-unfonts-core \
        ttf-wqy-zenhei \
        ttf-mscorefonts-installer \
        fonts-freefont-ttf \
        xvfb \
        curl \
    && apt-get clean \
    && apt-get autoremove -y \
    && rm -rf /var/lib/apt/lists/*

## TODO: randomize font installs and add from google https://gist.github.com/keeferrourke/d29bf364bd292c78cf774a5c37a791db

RUN mkdir -p /opt/bin \
    && chmod +x /dev/shm \
    && mkdir -p /brorun/browsers

RUN mkdir -p /app/brorun
COPY brorun /app/brorun/brorun
COPY .eslint* /app/brorun/
COPY .gitignore /app/brorun/
COPY .prettierrc /app/brorun/
COPY package.json yarn.lock /app/brorun/

WORKDIR /app/brorun

ENV BROWSERS_DIR /brorun/browsers

RUN cd /app/brorun \
    && yarn --frozen-lockfile \
    && npx install-browser-deps

RUN groupadd -r bro && useradd -r -g bro -G audio,video bro \
    && mkdir -p /home/bro/Downloads \
    && mkdir -p /home/bro/.cache/ulixee \
    && chown -R bro:bro /home/bro \
    && chown -R bro:bro /app/brorun \
    && chmod 777 /tmp \
    && chmod 777 /brorun/browsers \
    && chmod -R 777 /home/bro/.cache/ulixee

# Add below to run as unprivileged user.
USER bro

# NOTE: the above could be run as a base docker image for all brorun dockerfiles
#       the below code is specific for the playground deployment

EXPOSE 8080

# ensure to run code in production mode
ENV NODE_ENV=production

FROM base AS playground

RUN yarn workspace @brorun/playground build

# default cmd
CMD [ "yarn", "workspace", "@brorun/playground", "start" ]

FROM base AS server

RUN yarn workspace @brorun/server build

# default cmd
CMD [ "yarn", "workspace", "@brorun/server", "start" ]
```

which I build locally (CI/cloud one is a bit different):

```
docker build -t otainsight:brorun --target playground .
```

and which I run as follows:

```
docker run -p 8080:8080 otainsight:brorun
```

**Comment by blakebyrnes** [comment]
Ok, this actually helps a little. It's also a new error. Before it couldn't read the install_deps file. Now it's just telling you it didn't run. My guess is that either this command `yarn workspace @brorun/playground build` or this one `yarn workspace @brorun/server build` is the one actually prompting your system to install Chrome. 

I _think_ you might need two changes:
1. Whichever of `yarn workspace @brorun/server build` and `@brorun/playground build` is actually using Hero Code needs to add `&& npx install-browser-deps` to it.
2. You probably want to move the USER switch and chmods BELOW these installs

**Comment by GlenDC** [comment]
> Ok, this actually helps a little. It's also a new error.

No it 's the same old error. It's just that in previous attempts I tried to get around that by running that script as it asks, which is when you get into permission errors :)

**Comment by blakebyrnes** [comment]
> > Ok, this actually helps a little. It's also a new error.
> 
> No it 's the same old error. It's just that in previous attempts I tried to get around that by running that script as it asks, which is when you get into permission errors :)

Oh, the one you pasted above says: " EACCES: permission denied, open '/tmp/apt-install-chrome-dependencies.sh'", which is more indicative of a permissions error.

**Comment by blakebyrnes** [comment]
Maybe that's the runtime error. Do you get same runtime error?

**Comment by GlenDC** [comment]
Playground ans Server are different independent things, both use the core service which is what is using  the ulixee server and hero.

Locally this doesn't seem to be an issue though, as all my developers have no issue running it without local calls to `npx install-browser-deps`. But i suppose that is because my developers are on MacOS while on Linux things are bit different fo the ulixee projects, as we discussed before.

That said... I do already call `npx install-browser-deps`, as can be seen in my Dockerfile. And also already I do the user switch and CHMODS only below these instals, so I do not think the solution is there? At the point where I build the playground or server, I already installed the dependencies, so at that point it should no longer matter, right? That last build step just transpiles the typescript of my own prijects to javasript, cannot imagine that is in anyway related.

**Comment by blakebyrnes** [comment]
```
FROM base AS playground

RUN yarn workspace @brorun/playground build
```

You're calling a separate build process in a second "layer" that is AFTER you've already changed user. Is this doing any npm installs? That was my assumption

**Comment by blakebyrnes** [comment]
> Locally this doesn't seem to be an issue though, as all my developers have no issue running it without local calls to `npx install-browser-deps`. But i suppose that is because my developers are on MacOS while on Linux things are bit different fo the ulixee projects, as we discussed before.

This is only needed on Debian to install system dependencies. Mac Apps package everything up.

**Comment by blakebyrnes** [comment]
> ```
> FROM base AS playground
> 
> RUN yarn workspace @brorun/playground build
> ```
> 
> 
>      
>   
> 
> You're calling a separate build process in a second "layer" that is AFTER you've already changed user. Is this doing any npm installs? That was my assumption

Oh, sorry. I just saw you said it only transpires in that layer.

**Comment by GlenDC** [comment]
> You're calling a separate build process in a second "layer" that is AFTER you've already changed user. Is this doing any npm installs? That was my assumption

Nope, that is only doing the building of my selected TS project to JS, the installs (e.g. deps of ulixee) are already done in ase layer.

> This is only needed on Debian to install system dependencies. Mac Apps package everything up.

Yup that confirms what I wrote above and what we discussed before.

> Oh, sorry. I just saw you said it only transpires in that layer.

Yup

**Comment by blakebyrnes** [comment]
I'm a bit stumped. I don't understand why this would be working for me and not your setup. Do you think you could send me your yarn.lock file or copy out any of the references to @ulixee and @unblocked-web modules?

**Comment by GlenDC** [comment]
You ask, and I serve:

```
cat yarn.lock | grep -E 'ulixee|unblocked' | grep '"@' | sort | uniq | pbcopy
```

```
    "@ulixee/chrome-105-0" "^5195.127.6"
    "@ulixee/chrome-app" "^1.0.2"
    "@ulixee/chrome-app" "^1.0.3"
    "@ulixee/commons" "2.0.0-alpha.12"
    "@ulixee/crypto" "2.0.0-alpha.12"
    "@ulixee/databox" "2.0.0-alpha.12"
    "@ulixee/databox-core" "2.0.0-alpha.12"
    "@ulixee/databox-core-runtime" "2.0.0-alpha.12"
    "@ulixee/databox-for-hero" "2.0.0-alpha.12"
    "@ulixee/databox-for-hero-core-runtime" "2.0.0-alpha.12"
    "@ulixee/databox-interfaces" "2.0.0-alpha.12"
    "@ulixee/hero" "2.0.0-alpha.12"
    "@ulixee/hero-core" "2.0.0-alpha.12"
    "@ulixee/hero-interfaces" "2.0.0-alpha.12"
    "@ulixee/hero-plugin-utils" "2.0.0-alpha.12"
    "@ulixee/hero-timetravel" "2.0.0-alpha.12"
    "@ulixee/net" "2.0.0-alpha.12"
    "@ulixee/sidechain" "2.0.0-alpha.12"
    "@ulixee/specification" "2.0.0-alpha.12"
    "@unblocked-web/agent" "2.0.0-alpha.11"
    "@unblocked-web/agent-mitm" "2.0.0-alpha.11"
    "@unblocked-web/agent-mitm-socket" "2.0.0-alpha.11"
    "@unblocked-web/default-browser-emulator" "2.0.0-alpha.11"
    "@unblocked-web/default-human-emulator" "2.0.0-alpha.11"
    "@unblocked-web/js-path" "2.0.0-alpha.1"
    "@unblocked-web/real-user-agents" "2.0.0-alpha.11"
    "@unblocked-web/specifications" "2.0.0-alpha.11"
"@ulixee/chrome-105-0@^5195.127.6":
"@ulixee/chrome-98-0@^4758.109.6":
"@ulixee/chrome-app@^1.0.2", "@ulixee/chrome-app@^1.0.3":
"@ulixee/commons@2.0.0-alpha.12":
"@ulixee/crypto@2.0.0-alpha.12":
"@ulixee/databox-core-runtime@2.0.0-alpha.12":
"@ulixee/databox-core@2.0.0-alpha.12":
"@ulixee/databox-for-hero-core-runtime@2.0.0-alpha.12":
"@ulixee/databox-for-hero@2.0.0-alpha.12":
"@ulixee/databox-interfaces@2.0.0-alpha.12":
"@ulixee/databox@2.0.0-alpha.12":
"@ulixee/hero-core@2.0.0-alpha.12":
"@ulixee/hero-interfaces@2.0.0-alpha.12", "@ulixee/hero-interfaces@^2.0.0-alpha.12":
"@ulixee/hero-plugin-utils@2.0.0-alpha.12", "@ulixee/hero-plugin-utils@^2.0.0-alpha.12":
"@ulixee/hero-timetravel@2.0.0-alpha.12":
"@ulixee/hero@2.0.0-alpha.12", "@ulixee/hero@^2.0.0-alpha.11":
"@ulixee/net@2.0.0-alpha.12":
"@ulixee/server@^2.0.0-alpha.11":
"@ulixee/sidechain@2.0.0-alpha.12":
"@ulixee/specification@2.0.0-alpha.12":
"@unblocked-web/agent-mitm-socket@2.0.0-alpha.11":
"@unblocked-web/agent-mitm@2.0.0-alpha.11":
"@unblocked-web/agent@2.0.0-alpha.11":
"@unblocked-web/default-browser-emulator@2.0.0-alpha.11":
"@unblocked-web/default-human-emulator@2.0.0-alpha.11":
"@unblocked-web/js-path@2.0.0-alpha.1":
"@unblocked-web/real-user-agents@2.0.0-alpha.11":
"@unblocked-web/specifications@2.0.0-alpha.11", "@unblocked-web/specifications@^2.0.0-alpha.11":
```

**Comment by blakebyrnes** [comment]
I see two things standing out to me here:
1. You have Chrome 98 installed. That seems unexpected. I don't know where that would be coming from unless you have an old version of a library somewhere
2. The Ulixee libraries all stay lockstep (they're all on 2.0.0-alpha.12), while the unblocked is on 2.0.0-alpha.11. You have a few Ulixee libraries on an older version.

**Comment by blakebyrnes** [comment]
Total longshot, but if you delete yarn.lock and recreate it, does anything change in your docker?

**Comment by GlenDC** [comment]
Regenerated:

lock:

```
    "@ulixee/chrome-105-0" "^5195.127.6"
    "@ulixee/chrome-app" "^1.0.2"
    "@ulixee/chrome-app" "^1.0.3"
    "@ulixee/commons" "2.0.0-alpha.12"
    "@ulixee/crypto" "2.0.0-alpha.12"
    "@ulixee/databox" "2.0.0-alpha.12"
    "@ulixee/databox-core" "2.0.0-alpha.12"
    "@ulixee/databox-core-runtime" "2.0.0-alpha.12"
    "@ulixee/databox-for-hero" "2.0.0-alpha.12"
    "@ulixee/databox-for-hero-core-runtime" "2.0.0-alpha.12"
    "@ulixee/databox-interfaces" "2.0.0-alpha.12"
    "@ulixee/hero" "2.0.0-alpha.12"
    "@ulixee/hero-core" "2.0.0-alpha.12"
    "@ulixee/hero-interfaces" "2.0.0-alpha.12"
    "@ulixee/hero-plugin-utils" "2.0.0-alpha.12"
    "@ulixee/hero-timetravel" "2.0.0-alpha.12"
    "@ulixee/net" "2.0.0-alpha.12"
    "@ulixee/sidechain" "2.0.0-alpha.12"
    "@ulixee/specification" "2.0.0-alpha.12"
    "@unblocked-web/agent" "2.0.0-alpha.11"
    "@unblocked-web/agent-mitm" "2.0.0-alpha.11"
    "@unblocked-web/agent-mitm-socket" "2.0.0-alpha.11"
    "@unblocked-web/default-browser-emulator" "2.0.0-alpha.11"
    "@unblocked-web/default-human-emulator" "2.0.0-alpha.11"
    "@unblocked-web/js-path" "2.0.0-alpha.1"
    "@unblocked-web/real-user-agents" "2.0.0-alpha.11"
    "@unblocked-web/specifications" "2.0.0-alpha.11"
"@ulixee/chrome-105-0@^5195.127.6":
"@ulixee/chrome-98-0@^4758.109.6":
"@ulixee/chrome-app@^1.0.2", "@ulixee/chrome-app@^1.0.3":
"@ulixee/commons@2.0.0-alpha.12":
"@ulixee/crypto@2.0.0-alpha.12":
"@ulixee/databox-core-runtime@2.0.0-alpha.12":
"@ulixee/databox-core@2.0.0-alpha.12":
"@ulixee/databox-for-hero-core-runtime@2.0.0-alpha.12":
"@ulixee/databox-for-hero@2.0.0-alpha.12":
"@ulixee/databox-interfaces@2.0.0-alpha.12":
"@ulixee/databox@2.0.0-alpha.12":
"@ulixee/hero-core@2.0.0-alpha.12":
"@ulixee/hero-interfaces@2.0.0-alpha.12", "@ulixee/hero-interfaces@^2.0.0-alpha.12":
"@ulixee/hero-plugin-utils@2.0.0-alpha.12", "@ulixee/hero-plugin-utils@^2.0.0-alpha.12":
"@ulixee/hero-timetravel@2.0.0-alpha.12":
"@ulixee/hero@2.0.0-alpha.12", "@ulixee/hero@^2.0.0-alpha.11":
"@ulixee/net@2.0.0-alpha.12":
"@ulixee/server@^2.0.0-alpha.11":
"@ulixee/sidechain@2.0.0-alpha.12":
"@ulixee/specification@2.0.0-alpha.12":
"@unblocked-web/agent-mitm-socket@2.0.0-alpha.11":
"@unblocked-web/agent-mitm@2.0.0-alpha.11":
"@unblocked-web/agent@2.0.0-alpha.11":
"@unblocked-web/default-browser-emulator@2.0.0-alpha.11":
"@unblocked-web/default-human-emulator@2.0.0-alpha.11":
"@unblocked-web/js-path@2.0.0-alpha.1":
"@unblocked-web/real-user-agents@2.0.0-alpha.11":
"@unblocked-web/specifications@2.0.0-alpha.11", "@unblocked-web/specifications@^2.0.0-alpha.11":
```

**Comment by GlenDC** [comment]
However, same result I am afraid:

```
BrowserLaunchError: DependenciesMissingError: Ulixee Server failed to launch Chrome - Some of the dependencies needed to run Chrome 105.0.5195.125 are not on your system!

You can resolve this by running the apt dependency installer at:
-------------------------------------------------

/tmp/apt-install-chrome-dependencies.sh
```

**Comment by blakebyrnes** [comment]
Shoot. Do you have a direct link to @ulixee/server 2.0.0.11? Why is it still resolving to that version?

**Comment by blakebyrnes** [comment]
Also curious where the link to `"@ulixee/chrome-app" "^1.0.2"` coming from?

**Comment by blakebyrnes** [comment]
Ok. I'm pretty sure I know what this last piece is. You are running the npx command, but it only prints the command. It doesn't execute it. I probably should change that, but you can fix for now by doing '&& $(npx install-browser-deps)'

**Comment by GlenDC** [comment]
> Also curious where the link to `"@ulixee/chrome-app" "^1.0.2"` coming from?

`@ulixee/hero-core@2.0.0-alpha.12` > `"@unblocked-web/default-browser-emulator" "2.0.0-alpha.11"` > `"@ulixee/chrome-app" "^1.0.2"`

**Comment by GlenDC** [comment]
> Shoot. Do you have a direct link to @ulixee/server 2.0.0.11? Why is it still resolving to that version?

Yes I've now updated these manually to the latest (`alpha-12`)
